model/DataObjects.inc.php

```diff
@@ -648,6 +655,14 @@ public static function delete($libraryID, $key) {
 		self::unload($obj->id);
 		
 		if ($deleted) {
+			// If the attachment item was successfully deleted, check
+			// if other items are using this storageFileID,
+			// and if not, delete the row with storageFileID and libraryID from
+			// storageFileLibraries table
+			if ($type == 'item' && $obj->isImportedAttachment()) {
+				Zotero_Storage::deleteFileLibraryReference($storageFileID, $libraryID);
+			}
+			
```

Comment:
If we would move this before the item is deleted, then we also have to check if there are at least two file items left, because one will be removed. `SELECT COUNT(*)>1 FROM storageFileItems WHERE storageFileID = ? LIMIT 2`. Or do you mean just move it before `if ($deleted) {`